### DISCUSSION TITLE
Pop snackbar

### DISCUSSION THREAD

**Discussion by thegioiso** [bdlukaa/fluent_ui#630] [Q&A] [answered]
We had `fluent.showSnackbar();` to show up a snackbar. And i see that every snackbar will stack on other. Can i have a function like`fluent.popCurrentSnackbar()`  to pop it before we call show a new one?

**Comment by robkri** [comment, marked as answer]
If found this in the documentation of `showSnackbar`: 

To dismiss the snackbar manually, use the following code:
```dart
final result = showSnackbar(context, snackbar);
result.remove();
```

Is this what you are looking for?

EDIT:
To avoid an error when the Snackbar already got dismissed automatically, wrap the call of `remove()` in an try-catch-block like so: https://github.com/bdlukaa/fluent_ui/discussions/630#discussioncomment-4282055.

**Reply by thegioiso** [reply]
This is not what i need here, because it will cause error if the snack has dismissed.
Example:
We call a snack bar and its will auto dissmiss after 5s.
Before 5s, we can manually dismiss by this method.
But, after 5s, if we call this method, an error will occur:
```
══╡ EXCEPTION CAUGHT BY GESTURE ╞═══════════════════════════════════════════════════════════════════
The following assertion was thrown while handling a gesture:
Assertion failed:
D:\…\widgets\overlay.dart:161
_overlay != null
is not true
When the exception was thrown, this was the stack:
``` 
So... Maybe i will need another way....

Thank you so much for your support.

**Reply by thegioiso** [reply]
Finally, i put this method in `try catch` block, everything works just... fine...
```
 try {
        _currentSnackBar.remove();
        // ignore: empty_catches
      } catch (e) {}
      /// go show new snackbarrrrr
```

**Reply by robkri** [reply]
Nice. Maybe I'll need this, too, some time.

I updated my answer accordingly.
